opensrp-eusm/src/main/java/org/smartregister/eusm/repository/AppStructureRepository.java

```diff
@@ -272,7 +272,9 @@ public List<org.smartregister.domain.Location> getStructuresByDistrictId(String
         };
 
         String query = "SELECT " + StringUtils.join(columns, ",") + " from " + StructureRepository.STRUCTURE_TABLE
-                + " join location on location._id = " + StructureRepository.STRUCTURE_TABLE + ".parent_id ";
+                + " join location on location._id = " + StructureRepository.STRUCTURE_TABLE + ".parent_id"
+                + " AND " + StructureRepository.STRUCTURE_TABLE + "." + LATITUDE + " IS NOT NULL "
+                + " AND " + StructureRepository.STRUCTURE_TABLE + "." + LONGITUDE + " IS NOT NULL";
```

Comment:
Is this function used anywhere else?

Reply:
Is there a scenario where this function is called and lat, lon are optional?

Reply:
Yes, I checked that earlier. There is only 1 usage of this method, we can safely modify it.